kotlin/workflow-ui-android/src/main/java/com/squareup/workflow/ui/WorkflowRunner.kt

```diff
@@ -28,4 +44,161 @@ interface WorkflowRunner<out OutputT> {
+    }
+
+    /**
+     * Returns a [ViewModel][android.arch.lifecycle.ViewModel] implementation of
+     * [WorkflowRunner], tied to the given [fragment].
+     *
+     * It's probably more convenient to subclass [WorkflowFragment] rather than calling
+     * this method directly.
+     */
+    fun <InputT, OutputT : Any> of(
+      fragment: Fragment,
+      viewRegistry: ViewRegistry,
+      workflow: Workflow<InputT, OutputT, Any>,
+      inputs: Flowable<InputT>,
+      savedInstanceState: Bundle?
+    ): WorkflowRunner<OutputT> {
+      val factory =
+        WorkflowRunnerViewModel.Factory(workflow, viewRegistry, inputs, savedInstanceState)
+      @Suppress("UNCHECKED_CAST")
+      return ViewModelProviders.of(fragment, factory)[WorkflowRunnerViewModel::class.java]
+          as WorkflowRunner<OutputT>
+    }
+  }
+}
+
+/**
+ * Call this method from [FragmentActivity.onCreate], instead of [FragmentActivity.setContentView].
+ * It creates a [WorkflowRunner] for this activity, if one doesn't already exist, and
+ * sets a view driven by that model as the content view.
+ *
+ * Hold onto the [WorkflowRunner] returned and:
+ *
+ *  - Call [FragmentActivity.workflowOnBackPressed] from [FragmentActivity.onBackPressed] to allow
+ *    workflows to handle back button events. (See [HandlesBack] for more details.)
+ *
+ *  - Call [WorkflowRunner.onSaveInstanceState] from [FragmentActivity.onSaveInstanceState].
+ *
+ *  e.g.:
+ *
+ *     class MainActivity : AppCompatActivity() {
+ *       private lateinit var runner: WorkflowRunner<*, *>
+ *
+ *       override fun onCreate(savedInstanceState: Bundle?) {
+ *         super.onCreate(savedInstanceState)
+ *         runner = setContentWorkflow(MyViewRegistry, MyRootWorkflow(), savedInstanceState)
+ *       }
+ *
+ *       override fun onBackPressed() {
+ *         if (!runner.onBackPressed(this)) super.onBackPressed()
+ *       }
+ *
+ *       override fun onSaveInstanceState(outState: Bundle) {
+ *         super.onSaveInstanceState(outState)
+ *         runner.onSaveInstanceState(outState)
+ *       }
+ *     }
+ */
+@ExperimentalWorkflowUi
+@CheckResult
+fun <InputT, OutputT : Any> FragmentActivity.setContentWorkflow(
+  viewRegistry: ViewRegistry,
+  workflow: Workflow<InputT, OutputT, Any>,
+  inputs: Flowable<InputT>,
+  savedInstanceState: Bundle?
+): WorkflowRunner<OutputT> {
+  val runner = WorkflowRunner.of(this, viewRegistry, workflow, inputs, savedInstanceState)
+  val layout = WorkflowLayout(this@setContentWorkflow)
+      .apply {
+        id = R.id.workflow_layout
+        setWorkflowRunner(runner)
+      }
+
+  this.setContentView(layout)
+
+  return runner
+}
+
+/**
+ * If your workflow needs to manage the back button, override [FragmentActivity.onBackPressed]
+ * and call this method, and have your views or coordinators use [HandlesBack].
+ *
+ * e.g.:
+ *
+ *    override fun onBackPressed() {
+ *      if (!workflowViewModel.onBackPressed(this)) super.onBackPressed()
```

Comment:
Should this variable be named `runner` to be consistent with the previous code snippets?

Reply:
Yes, thanks, that's stale.